recipe/0014-Use-conda-s-tzdata-package.patch

```diff
@@ -0,0 +1,44 @@
+From 71951a875d0c006770816bd14bf5f6fc9043f8e1 Mon Sep 17 00:00:00 2001
+From: Isuru Fernando <isuruf@gmail.com>
+Date: Tue, 25 Jun 2024 13:06:14 -0500
+Subject: [PATCH] Use conda's tzdata package
+
+---
+ src/extra/tzone/localtime.c   | 6 +-----
+ src/library/base/R/datetime.R | 3 ++-
+ 2 files changed, 3 insertions(+), 6 deletions(-)
+
+diff --git a/src/extra/tzone/localtime.c b/src/extra/tzone/localtime.c
+index da2a99cac0..6a2bbae08a 100644
+--- a/src/extra/tzone/localtime.c
++++ b/src/extra/tzone/localtime.c
+@@ -472,11 +472,7 @@ tzload(const char * name, struct state * const sp, const int doextend)
+ 		p = "/var/db/timezone/zoneinfo";
+ #endif
+ 	    if (p == NULL || !strcmp(p, "internal")) {
+-		p = getenv("R_SHARE_DIR");
+-		if(p)
+-		    snprintf(buf, 1000, "%s/zoneinfo", p);
+-		else
+-		    snprintf(buf, 1000, "%s/share/zoneinfo", getenv("R_HOME"));
++		snprintf(buf, 1000, "%s/../../share/zoneinfo", getenv("R_HOME"));
+ 		buf[999] = '\0';
+ 		p = buf;
+ 	    }
+diff --git a/src/library/base/R/datetime.R b/src/library/base/R/datetime.R
+index f870e95a96..360232487e 100644
+--- a/src/library/base/R/datetime.R
++++ b/src/library/base/R/datetime.R
+@@ -81,7 +81,8 @@ Sys.timezone <- function(location = TRUE)
+     tzdir <- Sys.getenv("TZDIR")
+     if(nzchar(tzdir) && !dir.exists(tzdir)) tzdir <- ""
+     if(!nzchar(tzdir)) { ## See comments in OlsonNames
+-        if(dir.exists(tzdir <- "/usr/share/zoneinfo") ||
++        if(dir.exists(tzdir <- file.path(R.home(), "..", "..", "share", "zoneinfo")) ||
```

Comment:
There are a few places in `datetime.R` that reference `file.path(R.home("share"), "zoneinfo")` which we can/should probably substitute with `file.path(R.home("share"), "..", "..", "zoneinfo")`.

Reply:
Missed the one for macOS on line 57.

Reply:
No, that's intentional. It's only for checking internal version against system version.

Reply:
Yes, I saw that.
But I also don't see why we'd want to deviate there.
Yes, in almost all cases we'd expect ours to be recent.
But it's still the standard behavior on macOS to always use the system one if newer, isn't it?
As I read the code, if the user would install `r-base=4.4 tzdata=2020`, expected default be to use the system's one -- which would not be the case if we don't patch that line.

In case I'm missing something and there is good reason to deviate there, could you at least add a comment to explain it?

Reply:
>  But it's still the standard behavior on macOS to always use the system one if newer, isn't it?

No, that part is dead code anyway since we don't have a `VERSION` file in our tzdata package.

Reply:
Ah, I didn't think of that.
Could you still add a comment at least?
(Others/future selves shouldn't need to waste time checking those things.)

Reply:
ty